### PR TITLE
Add validation for password reset confirmation

### PR DESCRIPTION
**Pull request by bwndai**
The existing behavior of dj-rest-auth provided serializer PasswordResetConfirmSerializer validate the token and set the password in one go, this change extends the serializer and make that a two steps process.

### CODE REVIEW

File: src/planscape/planscape/settings.py
Comment (sterlingwellscaffeine) [inline src/planscape/planscape/settings.py:112]: According to the TODO message this needs to be removed or restructured into a settings.dev.py file.

File: src/planscape/planscape/urls.py
Comment (sterlingwellscaffeine) [inline src/planscape/planscape/urls.py:34]: Is this comment still relevant?

File: src/planscape/users/forms.py
Comment (sterlingwellscaffeine) [inline src/planscape/users/forms.py:11]: Please provide a class level comment describing the purpose of this class for future maintainers

File: src/planscape/users/serializers.py
Comment (sterlingwellscaffeine) [inline src/planscape/users/serializers.py:18]: Please add comments for these classes to describing their purpose for future maintainers.

**Review by sterlingwellscaffeine — APPROVED** [review]

File: src/planscape/planscape/settings.py
Comment (janicejl) [reply src/planscape/planscape/settings.py:112]: Rebasing to main should resolve this todo.
The TODO was resolved here: https://github.com/OurPlanscape/Planscape/pull/948

**Comment by runforyourlives** [comment]
With some of the desired changes lost in a bad git merge, we've added the diff to the original jira entry and will not merge this PR.

**Pull request closed without merging**
